File: Patrones/Creacionales/Builder/Python/ConstructorDeCasaConcreto.py
from ConstructorDeCasa import ConstructorDeCasa

class ConstructorDeCasaConcreto(ConstructorDeCasa):
    def construir_cimientos(self):
        self._casa.set_foundation("Cimientos de concreto")

    def construir_estructura(self):
        self._casa.set_estructura("Estructura de madera")

    def construir_techo(self):
        self._casa.set_techo("Techo de tejas")

    def construir_garage(self):
        self._casa.set_garage(True)

    def construir_piscina(self):
        self._casa.set_piscina(False)

    def construir_jardin(self):
        self._casa.set_jardin(True)

File: Patrones/Creacionales/Builder/Java/src/ConstructorDeCasaConcreto.java
public class ConstructorDeCasaConcreto extends ConstructorDeCasa {
    @Override
    public void construirCimientos() {
        casa.setFoundation("Cimientos de concreto");
    }

    @Override
    public void construirEstructura() {
        casa.setEstructura("Estructura de madera");
    }

    @Override
    public void construirTecho() {
        casa.setTecho("Techo de tejas");
    }

    @Override
    public void construirGarage() {
        casa.setGarage(true);  // Optional
    }

    @Override
    public void construirPiscina() {
        casa.setPiscina(false);  // Optional
    }

    @Override
    public void construirJardin() {
        casa.setJardin(true);  // Optional
    }
}
